src/components/RegisterForm.tsx

```diff
@@ -0,0 +1,118 @@
+'use client'
+
+import { useState } from 'react'
+import { useRouter } from 'next/navigation'
+export default function RegisterForm() {
+  const [name, setName] = useState('')
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState('')
+
+  const router = useRouter()
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+
+    try {
+      const resUserExists = await fetch('api/auth/userExists', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email }),
+      })
+
+      const { user } = await resUserExists.json()
+
+      if (user) {
+        setError('User already exists.')
+        return
+      }
+
+      const res = await fetch('api/auth/register', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name,
+          email,
+          password,
+        }),
+      })
+
+      if (res.ok) {
+        const form = e.target
+        form.reset()
+        router.push('/')
+      } else {
+        console.log('User registration failed.')
+      }
+    } catch (error) {
+      console.log('Error during registration: ', error)
+    }
+  }
+
+  const handleHaveAccountButton = () => {
+    router.push('/')
+  }
```

Comment:
User registration - Handles form submission, checks for existing users, and triggers registration.